src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -104,15 +108,39 @@ public AuthorizeFilter(string policy)
 
         bool IFilterFactory.IsReusable => true;
 
-        /// <inheritdoc />
-        public virtual async Task OnAuthorizationAsync(AuthorizationFilterContext context)
+        private async Task<AuthorizationPolicy> GetEffectivePolicyAsync(AuthorizationFilterContext context)
         {
-            if (context == null)
-            {
-                throw new ArgumentNullException(nameof(context));
+            if (_effectivePolicy != null) {
+                return _effectivePolicy;
             }
 
             var effectivePolicy = Policy;
+
+            if (_mvcOptions == null) {
+                _mvcOptions = context.HttpContext.RequestServices.GetRequiredService<IOptions<MvcOptions>>().Value;
+            }
+
+            if (_mvcOptions.CombineAuthorizeFilters)
```

Comment:
You seem to have lost the following check in this iteration?
```c#
if (!context.IsEffectivePolicy<AuthorizeFilter>(this)
{
      return;
}
```

Reply:
Looks like it, good catch, I'll see if I can add a test to ensure that only a single filter is calling into the authorize service

Reply:
Ok updated, plugged in a custom PolicyEvaluator that counts calls, and verified that the count was 2 before the fix, and now is 1 as expected.